From 9e8608dd48539772377b0232aefb8ea38d4b0a3e Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Thu, 3 Apr 2014 16:06:24 +0800
Subject: [PATCH 443/449] ENGR00306838 csi capture test: Close v4l2 device
 when it needn't

Some v4l2 device may open by csi capture test case that not support
capture function, close these device when it needn't.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 test/mxc_v4l2_test/csi_v4l2_capture.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/mxc_v4l2_test/csi_v4l2_capture.c b/test/mxc_v4l2_test/csi_v4l2_capture.c
index 37c185a..15157c7 100644
--- a/test/mxc_v4l2_test/csi_v4l2_capture.c
+++ b/test/mxc_v4l2_test/csi_v4l2_capture.c
@@ -198,6 +198,8 @@ static int find_video_device(void)
 		if (cap.capabilities & V4L2_CAP_VIDEO_CAPTURE) {
 			printf("Found v4l2 capture device %s.\n", v4l_device);
 			break;
+		} else {
+			close(fd_v4l);
 		}
 
 		i++;
-- 
1.8.0

